rfcs/debuggability_metrics.md

```diff
@@ -0,0 +1,126 @@
+
+## Polling
+
+While potentially less reliable than absolute metrics such as measuring time, polling customers for their preferences and pain points is still valuable. We would reach out to the sig-ux in order to develop questions. Measuring user preferences is not a primary skill set of our team and we would want to lean on experts to ensure an effective solution for our customers.
+
+In addition to broad polls, we will include deeper individual customer reports where we work with a single developer to examine their experience directly. For this step, we would work with a small sample of specific developers to get a detailed view of their debugging experience before and after changes. While this will take more time, it will provide more reliable results and allow us to guarantee accuracy in our polling.
+
+### Checks
+
+As a part of polling, we want to ensure that certain key checks are hit. These are:
+
+* Were you able to reproduce the bug locally, the first time?
+* Were you able to reproduce the bug locally?
+* Were you able to reproduce the bug?
+
+These are, in order, valuable benchmarks to hit. Reproduction of a bug is key in fixing it. Being able to reproduce it locally allows a developer to glean important information in fixing it. Finally, being able to reproduce it the first time is helpful in simply saving developer time by not forcing them to repeat steps.
+
+### Implementation
+
+* Write script that scrapes Jenkins build logs to measure Failure first signaled and Failure e-mail sent metrics
+* Perform external developer polling (this would be where most questions would be) (this would include log-finding, log-parsing, artifact-finding metrics)
+
+### Possible Outcomes / Scenarios
+
+Between metrics and polling, we will be able to generate a better picture of our pain sources and problems than we have now. While we won't be able to predict every possible outcome of these metrics and polling, we will attempt to identify several likely ones and comment briefly on our plans for those outcomes.
+
+Outcome/Scenario | Planned Steps
+-- | --
+Problems are not being logged due to unclear issue-filing process | Provide/improve documentation on issue-filing, run trainings on issue-filing
+Problems are not being logged due to effort required | Provide quick-links and other shortcuts, create issue templates
+Problems are being logged, but are not being actioned on | Use metrics to provide data to prioritize issues higher, reach out to appropriate owners
+
+## What are the advantages of the suggestion?
+
+* Provides absolute metrics in debugability (see table above)
+* Polls customers directly to get their experiences
+* Focuses on the key aspects of debugability and provides a limited scope
+* Low investment cost
```

Comment:
I'd suggest that even holding a single UX experiment is fairly high investment cost for obtaining metrics. And this may point to doing many of trials, both to track metrics over time and to increase data quality. I'm unsure how actionable we would find the results of a single standardized experiment.